lib/security/JweToken.ts

```diff
@@ -309,4 +298,68 @@ export default class JweToken extends JoseToken {
     // 18. If there was no recipient, the JWE is invalid. Otherwise output the plaintext.
     return plaintext.toString('utf8');
   }
+
+  /**
+   * Converts the JWE from the constructed type into a Compact JWE
+   */
+  public toCompactJwe (): string {
+    if (this.encryptedKey === undefined || this.payload === undefined || this.iv === undefined || this.aad === undefined || this.tag === undefined) {
+      throw new Error('Could not parse contents into a JWE');
+    }
+    const protectedHeaders = this.getProtectedHeader();
+    if (!('alg' in protectedHeaders) || !('enc' in protectedHeaders)) {
+      throw new Error("'alg' and 'enc' are required to be in the protected header");
+    }
+    if (this.aad.compare(Buffer.from(this.protectedHeaders || '')) !== 0) {
```

Comment:
Sanity: did you mean to compare to protected headers directly?

Reply:
The default AAD data (and required AAD data for compact JWEs) is the protected headers Base64url string. You would not be able to convert the JWE to compact without this condition being true.